felix/netlinkshim/netlink.go

```diff
@@ -47,6 +49,156 @@ type Interface interface {
+	nlHandle *netlink.Handle
+}
+
 func NewRealNetlink() (Interface, error) {
-	return netlink.NewHandle(syscall.NETLINK_ROUTE)
+	nlHandle, err := netlink.NewHandle(syscall.NETLINK_ROUTE)
+	return &RealNetlink{
+		nlHandle: nlHandle,
+	}, err
+}
+
+func (r *RealNetlink) SetSocketTimeout(to time.Duration) error {
+	return r.nlHandle.SetSocketTimeout(to)
+}
+
+func (r *RealNetlink) SetStrictCheck(b bool) error {
+	return r.nlHandle.SetStrictCheck(b)
+}
+
+func (r *RealNetlink) LinkList() ([]netlink.Link, error) {
+	retries := 3
+	for {
+		links, err := r.nlHandle.LinkList()
+		if err != nil {
+			if errors.Is(err, unix.EINTR) && retries > 0 {
+				retries--
+				continue
+			}
+		}
+		return links, err
+	}
+}
+
+func (r *RealNetlink) LinkByName(name string) (netlink.Link, error) {
+	return r.nlHandle.LinkByName(name)
+}
+
+func (r *RealNetlink) LinkAdd(link netlink.Link) error {
+	return r.nlHandle.LinkAdd(link)
+}
+
+func (r *RealNetlink) LinkDel(link netlink.Link) error {
+	return r.nlHandle.LinkDel(link)
+}
+
+func (r *RealNetlink) LinkSetMTU(link netlink.Link, mtu int) error {
+	return r.nlHandle.LinkSetMTU(link, mtu)
+}
+
+func (r *RealNetlink) LinkSetUp(link netlink.Link) error {
+	return r.nlHandle.LinkSetUp(link)
+}
+
+func (r *RealNetlink) RouteListFiltered(family int, filter *netlink.Route, filterMask uint64) ([]netlink.Route, error) {
+	retries := 3
+	for {
+		routes, err := r.nlHandle.RouteListFiltered(family, filter, filterMask)
+		if err != nil {
+			if errors.Is(err, unix.EINTR) && retries > 0 {
+				retries--
+				continue
+			}
+		}
+		return routes, err
+	}
+}
+
+func (r *RealNetlink) RouteAdd(route *netlink.Route) error {
+	return r.nlHandle.RouteAdd(route)
+}
+
+func (r *RealNetlink) RouteReplace(route *netlink.Route) error {
+	return r.nlHandle.RouteReplace(route)
+}
+
+func (r *RealNetlink) RouteDel(route *netlink.Route) error {
+	return r.nlHandle.RouteDel(route)
+}
+
+func (r *RealNetlink) AddrList(link netlink.Link, family int) ([]netlink.Addr, error) {
+	retries := 3
+	for {
+		addrs, err := r.nlHandle.AddrList(link, family)
+		if err != nil {
+			if errors.Is(err, unix.EINTR) && retries > 0 {
+				retries--
+				continue
+			}
+		}
+		return addrs, err
+	}
+}
+
+func (r *RealNetlink) AddrAdd(link netlink.Link, addr *netlink.Addr) error {
+	return r.nlHandle.AddrAdd(link, addr)
+}
+
+func (r *RealNetlink) AddrDel(link netlink.Link, addr *netlink.Addr) error {
+	return r.nlHandle.AddrDel(link, addr)
+}
+
+func (r *RealNetlink) RuleList(family int) ([]netlink.Rule, error) {
+	retries := 3
+	for {
+		rules, err := r.nlHandle.RuleList(family)
+		if err != nil {
+			if errors.Is(err, unix.EINTR) && retries > 0 {
+				retries--
+				continue
+			}
+		}
+		return rules, err
+	}
+}
+
+func (r *RealNetlink) RuleAdd(rule *netlink.Rule) error {
+	return r.nlHandle.RuleAdd(rule)
+}
+
+func (r *RealNetlink) RuleDel(rule *netlink.Rule) error {
+	return r.nlHandle.RuleDel(rule)
+}
+
+func (r *RealNetlink) Delete() {
+	//nolint:staticcheck
```

Comment:
What's the linter's problem?

Reply:
```netlink.Delete``` is deprecated and I got the below error from the linter. 
```SA1019: r.nlHandle.Delete is deprecated: use Close instead which is in line with typical resource release patterns for files and other resources.```

Reply:
Delete just calls through to Close so let's just switch to using Close?